packages/api/src/external/carequality/organization-fhir-template.ts

```diff
@@ -0,0 +1,285 @@
+            code: role,
+          },
+        ],
+      },
+    ],
+    contact: [
+      {
+        purpose: {
+          coding: [
+            {
+              system: "http://terminology.hl7.org/CodeSystem/contactentity-type",
+              code: "OPERATIONS",
+            },
+          ],
+        },
+        name: {
+          use: "official",
+          text: contactName,
+        },
+        telecom: [
+          {
+            system: "phone",
+            value: phone,
+          },
+          {
+            system: "email",
+            value: email,
+          },
+        ],
+        address: {
+          use: "work",
+          type: "both",
+          line: [addressLine1],
+          city,
+          postalCode,
+          country: "USA",
+        },
+      },
+    ],
+    extension: [
+      {
+        url: "https://sequoiaproject.org/fhir/sphd/StructureDefinition/PurposesOfUse",
+        valueCodeableConcept: {
+          coding: [
+            {
+              system: "http://healthit.gov/nhin/purposeofuse",
+              code: createPurposeOfUse(),
+            },
+          ],
+        },
+      },
+      {
+        url: "https://sequoiaproject.org/fhir/sphd/StructureDefinition/Domains",
+        valueCoding: {
+          system: "https://sequoiaproject.org/fhir/sphd/CodeSystem/Domains",
+          code: "CQ",
+        },
+      },
+    ],
+    address: [
+      {
+        use: "work",
+        type: "both",
+        line: [addressLine1],
+        city,
+        state,
+        postalCode,
+        country: "USA",
+        extension: [
+          {
+            url: "https://sequoiaproject.org/fhir/sphd/StructureDefinition/OrgLocation",
+            valueReference: {
+              reference: "#location-1",
+            },
+          },
+        ],
+      },
+    ],
+    partOf: {
+      reference: `Organization/${parentOrgOid ?? metriportOid}`,
+      type: "Organization",
+    },
+    endpoint: endpoints.map((_, index) => {
+      return {
+        reference: `#endpoint-${index + 1}`,
+        type: "Endpoint",
+      };
+    }),
+    contained: [
+      {
+        id: "location-1",
+        resourceType: "Location",
+        meta: {
+          profile: ["https://sequoiaproject.org/fhir/sphd/StructureDefinition/SequoiaLocation"],
+        },
+        position: {
+          longitude: parseFloat(lon),
+          latitude: parseFloat(lat),
+        },
+      },
+      ...endpoints.map((endpoint, index) => {
+        return {
+          id: `endpoint-${index + 1}`,
+          ...endpoint,
+        };
+      }),
+    ],
+  };
+  return org;
+}
+
+function getEndpoint(urnOid: string, urlType: ChannelUrl, url: string): Endpoint {
+  const channelType = urlType === XCPD_STRING ? "ihe-xcpd" : "ihe-xca";
+  const channelName =
+    urlType === XCPD_STRING
+      ? "Patient Discovery"
+      : urlType === XCA_DQ_STRING
+      ? "Query for Documents"
+      : "Retrieve Documents";
```

Comment:
Just to confirm, it can be only one of the three options?

Reply:
This is the old logic, but yes I guess